tests/utils/test_name_utils.py

```diff
@@ -0,0 +1,22 @@
+from mlflow.utils.name_utils import _generate_random_name, _generate_unique_integer_id
+
+
+def test_random_name_generation():
+
+    # Validate exhausted loop truncation
+    name = _generate_random_name(max_length=8)
+    assert len(name) == 8
+
+    # Validate default behavior while calling 1000 times that names end in integer
+    names = [_generate_random_name() for _ in range(1000)]
+    assert all(len(name) <= 20 for name in names)
```

Comment:
```suggestion
    assert all(len(name) == 18 for name in names)
```

Can we make this more strict?